### PR TITLE
add .github/CODEOWNERS

### PR DESCRIPTION
**Pull request by themr0c**
Adding a CODEOWNSERS file would help GitHub propose more accurate reviewers when you open a pull request.

Currently a draft, to check if it is something we can implement.
To make it work, engineers require write permissions on the repository.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (kim-tsao) [inline .github/CODEOWNERS:9]:
```suggestion
/modules/authentication @redhat-developer/rhdh-content @hmanwani-rh @themr0c @josephca @kim-tsao @JessicaJHee @04kash 
```
The security WG is handling a lot of the auth providers today, so let's get them added as codeowners

**Comment by nickboldt** [comment]
On hold for a couple reasons:
* team reorg is happening Real Soon Now (next sprint?)
* creation/auditing of existing rhd/rhdh groups needs to happen to clean things up, make them consistent, and document maintenance steps

Once the above is sorted we can switch this PR to use groups of SMEs instead of individual SMEs.

**Review by pabel-rh — APPROVED** [review]

**Pull request closed without merging**
